### PR TITLE
tiltfile2: basic test for loading a docker compose manifest

### PR DESCRIPTION
**Pull request by jazzdan**
@maiamcc any other assertions I should add for this basic test?

I also changed the docker compose yaml path to be absolute to match how we treat other files that we read and store in memory.

### CODE REVIEW

File: internal/tiltfile2/docker_compose.go [hunk 1]
```diff
@@ -23,17 +24,21 @@ func (s *tiltfileState) dockerCompose(thread *skylark.Thread, fn *skylark.Builti
 	if err != nil {
 		return nil, err
 	}
+	absConfigPath, err := filepath.Abs(configPath)
```
Comment (nicks) [inline internal/tiltfile2/docker_compose.go]: maybe ospath.RealAbs, like we did elsewhere?

Comment (jazzdan) [reply internal/tiltfile2/docker_compose.go]: yes, thank you!

File: internal/tiltfile2/tiltfile_test.go [hunk 2]
```diff
@@ -614,7 +614,24 @@ k8s_resource('foo', 'foo.yaml')
 }
 
 func TestDockerComposeResourceCreation(t *testing.T) {
-	// TODO(maia)
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
```
Comment (maiamcc) [inline internal/tiltfile2/tiltfile_test.go]: this is a buncha k8s-specific setup, i'm concerned it muddies the test. maybe just specifically set up the dc-specific bits you need?

File: internal/tiltfile2/tiltfile_test.go [hunk 3]
```diff
@@ -614,7 +614,24 @@ k8s_resource('foo', 'foo.yaml')
...
+    build: ./foo
+    command: sleep 100
+    ports:
+      - "12312:12312"`)
+	YAMLPath := f.TempDirFixture.JoinPath("docker-compose.yml")
+	tf := fmt.Sprintf("docker_compose('%s')", YAMLPath)
+	f.file("Tiltfile", tf)
+
+	f.load("foo")
+	f.assertManifest("foo", dcYAMLPath(YAMLPath))
```
Comment (maiamcc) [inline internal/tiltfile2/tiltfile_test.go]: nice. could you add a todo for me? -- assert Dockerfile and docker-compose.yml are config files

**Comment by maiamcc** [comment]
hmm do we want a docker compose client that we can mock out for tests so we don't actually have to call dc?

**Comment by jazzdan** [comment]
@maiamcc in my opinion not right now- we only use `docker-compose` here to read the YAML, not do any execution. It's similar to how this test suite depends on `kustomize`. But probably in the Upper we'll want a mock, yeah.

**Comment by maiamcc** [comment]
> @maiamcc in my opinion not right now- we only use docker-compose here to read the YAML, not do any execution. It's similar to how this test suite depends on kustomize. But probably in the Upper we'll want a mock, yeah.

`docker-compose config` VALIDATES the config as well as reading it so it does rely on our test fixture having a valid DC setup, which is maybe not ideal / muddies the test / makes set up more of a pain? but also eh, i agree that we can wait til we need it.

**Pull request merged**